wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -285,10 +285,11 @@ def _etag_from_obj(
         ]
     ) -> ETag:
         etag: ETag
-        if hasattr(obj, "e_tag"):
-            etag = obj.e_tag[1:-1]  # escape leading and trailing quote
+        if hasattr(obj, "meta") and obj.meta.data is None:
```

Comment:
This is better than checking for `hasattr(obj, "load")` since s3's `ObjectSummary` doesn't have a load() method. This check takes care of all three cases - Object, ObjectVersion and ObjectSummary.

Reply:
this feels really fragile, let chat offline on how to make thsi check better. I would rather have 2 methods if we have to than relying on this internal detail